src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -334,58 +330,84 @@ SignalrCoreTransport.prototype.updateQuery = function(
     });
 
     this.contextId = contextId;
-    this.authToken = authToken;
+    this.authToken = authToken.replace('BEARER ', '');
 
     if (forceAuth) {
-        this.renewSession(authToken, contextId);
+        this.renewSession(this.authToken, contextId);
     }
 };
 
 SignalrCoreTransport.prototype.renewSession = function(authToken, contextId) {
-    const options = {
-        headers: {
-            Authorization: authToken,
-        },
-    };
-    const url = `${this.baseUrl}/streaming/renewal/renewsession`;
+    if (!this.connection || this.isReconnecting) {

```

Comment:
if a token expires whilst we are connecting or reconnecting, don't we need to remember to renew the session once we are connected?

Reply:
we will reconnect with new token so I don't think thats needed 
I will still double check it

Reply:
there was issue with token factory
now fixed
so while reconnecting it will use new token

Reply:
is this because the token is sent right at the end of connecting?

Reply:
Not sure what you mean by that
It uses token factory i.e. a normal function which returns updated token
lets consider scenario of reconnecting state
signalr is trying to reconnect and we got new token and trying to renew session
we store this token as instance variable. The same token will be served from token factory. so next time signalr tries to reconnect it will use this token factory to get updated token
since we reconnected with new token we don't need to renew it again.